src/main/java/net/atos/zac/mail/MailService.java

```diff
@@ -130,33 +148,73 @@ public String sendMail(final MailGegevens mailGegevens, final Bronnen bronnen) {
         enkelvoudigInformatieobjectWithInhoud.setCreatiedatum(LocalDate.now());
-        enkelvoudigInformatieobjectWithInhoud.setTitel(onderwerp);
+        enkelvoudigInformatieobjectWithInhoud.setTitel(mailGegevens.getOnderwerp());
         enkelvoudigInformatieobjectWithInhoud.setAuteur(loggedInUserInstance.get().getFullName());
         enkelvoudigInformatieobjectWithInhoud.setTaal(ConfiguratieService.TAAL_NEDERLANDS);
         enkelvoudigInformatieobjectWithInhoud.setInformatieobjecttype(eMailObjectType.getUrl());
-        enkelvoudigInformatieobjectWithInhoud.setInhoud(body.getBytes(StandardCharsets.UTF_8));
+        enkelvoudigInformatieobjectWithInhoud.setInhoud(pdfDocument);
         enkelvoudigInformatieobjectWithInhoud.setVertrouwelijkheidaanduiding(Vertrouwelijkheidaanduiding.OPENBAAR);
-        enkelvoudigInformatieobjectWithInhoud.setFormaat(MediaType.TEXT_HTML);
-        enkelvoudigInformatieobjectWithInhoud.setBestandsnaam(String.format("%s.html", onderwerp));
+        enkelvoudigInformatieobjectWithInhoud.setFormaat(MEDIA_TYPE_PDF);
+        enkelvoudigInformatieobjectWithInhoud.setBestandsnaam(String.format("%s.pdf", mailGegevens.getOnderwerp()));
         enkelvoudigInformatieobjectWithInhoud.setStatus(InformatieobjectStatus.DEFINITIEF);
         enkelvoudigInformatieobjectWithInhoud.setVertrouwelijkheidaanduiding(Vertrouwelijkheidaanduiding.OPENBAAR);
         enkelvoudigInformatieobjectWithInhoud.setVerzenddatum(LocalDate.now());
         return enkelvoudigInformatieobjectWithInhoud;
     }
 
+    private byte[] createPdfDocument(final MailGegevens mailGegevens, final List<Attachment> attachments) {
+        final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        final Document document = new Document();
+        try {
+            PdfWriter.getInstance(document, byteArrayOutputStream);
+            document.open();
+            document.addTitle(mailGegevens.getOnderwerp());
+            final Font font = FontFactory.getFont(FontFactory.COURIER, 16, BaseColor.BLACK);
+            final Paragraph paragraph = new Paragraph(StringUtils.EMPTY, font);
+            addToParagraph(paragraph, MAIL_ONTVANGER, mailGegevens.getOntvanger().getEmail());
+            if (!attachments.isEmpty()) {
+                addToParagraph(paragraph, MAIL_BIJLAGE,
+                               attachments.stream().map(attachment -> String.valueOf(attachment.getFilename()))
+                                       .collect(Collectors.joining(", ")));
+            }
+            addToParagraph(paragraph, MAIL_ONDERWERP, mailGegevens.getOnderwerp());
+            paragraph.add(MAIL_BERICHT);
+            final HtmlCleaner htmlCleaner = new HtmlCleaner();
+            final TagNode root = htmlCleaner.clean(mailGegevens.getBody());
+            final String bodyHtml = String.format("<%s>%s</%s>", root.getName(), htmlCleaner.getInnerHtml(root),
```

Comment:
Ik kan het niet vinden in de javadoc maar logisch gezien: De innerHtml is HTML dus dat is veilig, maar die name zal decoded zijn, dus als je op deze manier een tag gaat maken zul je die strikt genomen weer moeten encoden (minimaal de 5 xml entities).

Mooier zou het zijn als er een nettere manier was om een tag te maken, want code maken door zelf strings te concateneren is altijd gevaarlijk als de strings die je gebruikt niet hard gecodeerd zijn.

Reply:
Heb er even naar gekeken.

En heb er een xml serializer tussen gezet. Dat werkt net wat schoner, zo kun je de volledige html uit de cleaner halen zonder dat je zelf nog de root tags toe moet voegen.